lib/auth/init.go

```diff
@@ -1048,12 +1048,23 @@ type PresetUsers interface {
 	UpsertUser(ctx context.Context, user types.User) (types.User, error)
 }
 
+// getPresetUsers returns a list of all users roles expected to be available on
+// this cluster.
+func getPresetUsers() []types.User {
+	presets := []types.User{
+		services.NewSystemAutomaticAccessBotUser(),
+	}
+
+	// Certain `New$FooUser()` functions will return a nil role if the
+	// corresponding feature is disabled. They should be filtered out as they
+	// are not actually made available on the cluster.
+	return slices.DeleteFunc(presets, func(r types.User) bool { return r == nil })
```

Comment:
Maybe? `r` seems out of nowhere.

```suggestion
	return slices.DeleteFunc(presets, func(u types.User) bool { return u == nil })
```